tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -0,0 +1,160 @@
+// <copyright file="ReturnedHeadersAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Specialized;
+using System.Net;
+using System.Net.Mime;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+#endif
+using static Datadog.Trace.Telemetry.Metrics.MetricTags;
+
+#nullable enable
+namespace Datadog.Trace.Iast;
+
+internal static class ReturnedHeadersAnalyzer
+{
+    private const string ContentTypeLow = "content-type";
+    private const string XContentTypeOptionsLow = "x-content-type-options";
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ReturnedHeadersAnalyzer));
+
+#if NETFRAMEWORK
+    internal static void Analyze(NameValueCollection headers, IntegrationId integrationId, string serviceName, int responseCode)
+    {
+        try
+        {
+            if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+            {
+                return;
+            }
+
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+
+            string contentTypeValue = string.Empty;
+            string contentOptionValue = string.Empty;
+
+            // We iterate instead of trying to get the key directly because keys are case insensitive
+            foreach (var headerKey in headers.AllKeys)
+            {
+                var headerKeyLow = headerKey.ToLowerInvariant();
+                if (headerKeyLow == ContentTypeLow)
+                {
+                    contentTypeValue = headers[headerKey];
+
+                    if (!IsHtmlResponse(contentTypeValue))
```

Comment:
A question for the RFC, but AFAIK `nosniff` should be set in various other contexts too, like JS?

Reply:
AFAIK, nosniff is only used in 'X_Content-Type-Options' response headers

Reply:
Sorry, I mean yes, it's only used in `X-Content-Type-Options` but AFAIK, `X-Content-Type-Options` is not only used on HTML...

Reply:
Yes, it can be used for other contents. The vulnerability is only considered for html types in IAST, though. It would be a good topic to discuss in the next IAST meeting. I will bring it.